src/installer/managed/Microsoft.NET.HostModel/AppHost/BinaryUtils.cs

```diff
@@ -173,5 +173,31 @@ public static void CopyFile(string sourcePath, string destinationPath)
             // Copy file to destination path so it inherits the same attributes/permissions.
             File.Copy(sourcePath, destinationPath, overwrite: true);
         }
+
+        internal static void SaveFile(MemoryMappedViewAccessor sourceViewAccessor, string destinationFilePath)
+        {
+            using FileStream fileStream = new FileStream(destinationFilePath, FileMode.Create);
+
+            int pos = 0;
+            int bufSize = 16384; //16K
+
+            byte[] buf = new byte[bufSize];
```

Comment:
is ArrayPool available?

Reply:
Yes, though I purposely did not use it here.  This code is invoked from MSBuild once per-process, and the only existing use of `ArrayPool` that I've been able to find is from `RegularExpressions`, which allocations `char[]`.  My goal here is to avoid incurring the cost of initializing an `ArrayPool<byte>` when we only need one byte[] in this code for the life of the process.